core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -646,20 +647,18 @@ public static function searchFor($strKeywords, $blnOrSearch=false, $arrPid=array
 	 *
 	 * @param string $strUrl The URL to be removed
 	 */
-	public static function removeEntry($strUrl)
+	public static function removeEntry($strUrl, Connection $connection = null)
 	{
-		$objDatabase = Database::getInstance();
+		/** @var Connection $connection */
+		$connection = $connection ?? System::getContainer()->get('database_connection');
 
-		$objResult = $objDatabase->prepare("SELECT id FROM tl_search WHERE url=?")
-								 ->execute($strUrl);
+		$stmt = $connection->prepare('SELECT id FROM tl_search WHERE url=:url');
+		$stmt->execute(array('url' => $strUrl));
 
-		while ($objResult->next())
+		foreach ($stmt->fetchAllAssociative() as $id)
```

Comment:
This causes an error `An exception occurred while executing 'DELETE FROM tl_search WHERE id = ?' with params [{"id":7}]`

```suggestion
		foreach ($stmt->fetchFirstColumn() as $id)
```

Missed that in my first review 🙈